services/person.ts

```diff
@@ -0,0 +1,196 @@
+import { querySudo, updateSudo } from '@lblod/mu-auth-sudo';
+import { sparqlEscapeUri } from 'mu';
+
+import { Conflict } from '../types';
+import { CustomError } from '../utils/custom-error';
+
+export async function getConflictingPersons(
+  batchSize: number | undefined,
+): Promise<Array<Conflict>> {
+  const limit = batchSize ? `LIMIT ${batchSize}` : '';
+  const queryString = `
+    PREFIX adms: <http://www.w3.org/ns/adms#>
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX person: <http://www.w3.org/ns/person#>
+    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+
+    SELECT DISTINCT ?person ?conflict
+    WHERE {
+      GRAPH ?g {
+        ?person a person:Person .
+        ?person adms:identifier / skos:notation ?rrn .
+      } 
+      ?g ext:ownedBy ?organization .
+
+      GRAPH ?g2 {
+        ?conflict a person:Person .
+        ?conflict adms:identifier / skos:notation ?rrn .
+
+        OPTIONAL {
+          ?conflict ext:conflictsWith ?conflictsWith .
+        }
+        FILTER(!BOUND(?conflictsWith) || ?conflictsWith = false)
+      } 
+      ?g2 ext:ownedBy ?organization2 .
+
+      FILTER(?person < ?conflict)
+    } ${limit}
+  `;
+  try {
+    const queryResult = await querySudo(queryString);
+    const bindings = queryResult.results?.bindings;
+
+    if (!bindings || bindings.length === 0) {
+      return null;
+    }
+
+    return bindings.map((b) => {
+      return {
+        personUri: b.person.value,
+        conflictUri: b.conflict.value,
+      };
+    });
+  } catch (error) {
+    throw new CustomError(
+      'Something went wrong while querying the person conflicts',
+    );
+  }
+}
+
+export async function getConflictingPersonUris(
+  conflicts: Array<Conflict>,
+): Promise<Array<Conflict>> {
+  if (conflicts.length === 0) {
+    return;
+  }
+
+  const values = conflicts.map(
+    (b) =>
+      `( ${sparqlEscapeUri(b.conflictUri)} ${sparqlEscapeUri(b.personUri)} )`,
+  );
+  const queryString = `
+    PREFIX persoon: <http://data.vlaanderen.be/ns/persoon#>
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX adms: <http://www.w3.org/ns/adms#>
+    PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+    PREFIX dct: <http://purl.org/dc/terms/>
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX person: <http://www.w3.org/ns/person#>
+    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+
+    SELECT DISTINCT ?person ?conflict ?isConflicting
```

Comment:
naming is a bit confusing. the other person with the same rrn is called conflict but it's possibly not conflicting?
i know what you mean but you've got to admit that we will both have forgotten in 3 weeks right?

Reply:
maybe isConflicting can be called 'unresolvable'?